src_rebuild/GAME/C/MISSION.C

```diff
@@ -1187,19 +1192,19 @@ int Swap2Cars(int curslot, int newslot)
 	do {
 		p_Var7 = p_Var2;
 		p_Var3 = cp;
-		uVar8 = *(undefined4 *)((p_Var7->hd).where.m + 2);
-		uVar11 = *(undefined4 *)((p_Var7->hd).where.m + 4);
-		uVar13 = *(undefined4 *)((p_Var7->hd).where.m + 6);
-		*(undefined4 *)(p_Var3->hd).where.m = *(undefined4 *)(p_Var7->hd).where.m;
-		*(undefined4 *)((p_Var3->hd).where.m + 2) = uVar8;
-		*(undefined4 *)((p_Var3->hd).where.m + 4) = uVar11;
-		*(undefined4 *)((p_Var3->hd).where.m + 6) = uVar13;
+		uVar8 = *(unsigned int *)((p_Var7->hd).where.m + 2);
+		uVar11 = *(unsigned int *)((p_Var7->hd).where.m + 4);
+		uVar13 = *(unsigned int *)((p_Var7->hd).where.m + 6);
+		*(unsigned int *)(p_Var3->hd).where.m = *(unsigned int *)(p_Var7->hd).where.m;
+		*(unsigned int *)((p_Var3->hd).where.m + 2) = uVar8;
+		*(unsigned int *)((p_Var3->hd).where.m + 4) = uVar11;
+		*(unsigned int *)((p_Var3->hd).where.m + 6) = uVar13;
```

Comment:
Matrix access should not be like direct memory access but instead use [r][c] access